tiledb/api/c_api/current_domain/current_domain_api_external_experimental.h

```diff
@@ -0,0 +1,185 @@
+/**
+ * @file
+ * tiledb/api/c_api/current_domain/current_domain_api_external_experimental.h
+ *
+ * @section LICENSE
+ *
+ * The MIT License
+ *
+ * @copyright Copyright (c) 2024 TileDB, Inc.
+ *
+ * Permission is hereby granted, free of charge, to any person obtaining a copy
+ * of this software and associated documentation files (the "Software"), to deal
+ * in the Software without restriction, including without limitation the rights
+ * to use, copy, modify, merge, publish, distribute, sublicense, and/or sell
+ * copies of the Software, and to permit persons to whom the Software is
+ * furnished to do so, subject to the following conditions:
+ *
+ * The above copyright notice and this permission notice shall be included in
+ * all copies or substantial portions of the Software.
+ *
+ * THE SOFTWARE IS PROVIDED "AS IS", WITHOUT WARRANTY OF ANY KIND, EXPRESS OR
+ * IMPLIED, INCLUDING BUT NOT LIMITED TO THE WARRANTIES OF MERCHANTABILITY,
+ * FITNESS FOR A PARTICULAR PURPOSE AND NONINFRINGEMENT. IN NO EVENT SHALL THE
+ * AUTHORS OR COPYRIGHT HOLDERS BE LIABLE FOR ANY CLAIM, DAMAGES OR OTHER
+ * LIABILITY, WHETHER IN AN ACTION OF CONTRACT, TORT OR OTHERWISE, ARISING FROM,
+ * OUT OF OR IN CONNECTION WITH THE SOFTWARE OR THE USE OR OTHER DEALINGS IN
+ * THE SOFTWARE.
+ *
+ * @section DESCRIPTION
+ *
+ * This file declares the current_domain experimental C API for TileDB.
+ */
+
+#ifndef TILEDB_CAPI_CURRENT_DOMAIN_API_EXTERNAL_EXPERIMENTAL_H
+#define TILEDB_CAPI_CURRENT_DOMAIN_API_EXTERNAL_EXPERIMENTAL_H
+
+#include "tiledb/api/c_api/api_external_common.h"
+#include "tiledb/api/c_api/context/context_api_external.h"
+#include "tiledb/api/c_api/ndrectangle/ndrectangle_api_external_experimental.h"
+
+#ifdef __cplusplus
+extern "C" {
+#endif
+
+/** The current domain type */
+typedef enum {
+#define TILEDB_CURRENT_DOMAIN_TYPE_ENUM(id) TILEDB_##id
+#include "tiledb/sm/c_api/tiledb_enum.h"
+#undef TILEDB_CURRENT_DOMAIN_TYPE_ENUM
+} tiledb_current_domain_type_t;
+
+typedef struct tiledb_current_domain_handle_t tiledb_current_domain_t;
+
+/**
+ * Create a current domain object
+ *
+ * **Example:**
+ *
+ * @code{.c}
+ * tiledb_current_domain_t *current_domain;
+ * tiledb_current_domain_create(ctx, &current_domain);
+ * tiledb_current_domain_free(&current_domain);
+ * @endcode
+ *
+ * @param ctx The TileDB context
+ * @param current_domain The current domain to be allocated
+ * @return `TILEDB_OK` for success and `TILEDB_ERR` for error.
+ */
+TILEDB_EXPORT int32_t tiledb_current_domain_create(
```

Comment:
Don't we also need the type in this constructor? I know we only have one now, but if we add more we will need to create a new constructor

Reply:
The type can be deducted in the current API whilst setting the shape, that is there will be an API call for setting each type of shape. If we tighten this in the near future, indeed the create function will need to take in the type and the rectangle itself.